upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -134,6 +134,29 @@ func (tf *TemplateFunctions) AddTo(dest template.FuncMap, secretStore fi.SecretS
 		dest["WeaveSecret"] = func() string { return weavesecretString }
 	}
 
+	if tf.cluster.Spec.Networking != nil && tf.cluster.Spec.Networking.Calico != nil {
+		flexvolumeDriverHost := ""
+		// TODO: have available distro here to select the right flexvolume path
+		switch "" {
```

Comment:
@justinsb Would it be possible to have in here the `Distribution`?

I've found out that it is available in the NodeUp Model but not in the Kops Model.

Does it make sense to have it in there?

Reply:
According to https://github.com/kubernetes/kops/blob/master/docs/development/how_it_works.md#4-cloudup, it does not make much sense to have in **cloudup** the distro because it is available in the *nodeup*.

Need to find another way.

Reply:
The only way I find at the moment to customize the dir is to add a new property in the `Calico` networking Spec https://github.com/kubernetes/kops/blob/master/pkg/apis/kops/networking.go#L79 like `VolumePluginDirectory` same as for the `KubeletSpec` https://github.com/kubernetes/kops/blob/master/pkg/apis/kops/componentconfig.go#L150

Reply:
So it seems like the issue is that cloudup is building this DaemonSet manifest that applies to the entire cluster, but you're wanting to reference a distro which is specific to the node (or InstanceGroup). One might have some FlatCar nodes and some CoreOS or ContainerOS nodes, and they would need their `calico-node` pods to have whichever hostPath is appropriate. I'm not sure Kops is able to support anything like that currently. Would nodeup adding a symlink be acceptable?

Reply:
Thank you @rifelpet to react on this ;-)

yeah you got the point.

what do you have in mind exactly regarding the symlink? Could you elaborate?

Reply:
hm it may be more difficult than I thought. We can't conditionally add a symlink in FlatCar from `/usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds` to `/var/lib/kubelet/volumeplugins/nodeagent~uds` because `/usr` is read-only as you point out. We wouldn't be able to have the DaemonSet use `/var/lib/kubelet/...` and symlink it back to `/usr/libexec/...` if the distro is *not* FlatCar because I think that rolling the DaemonSet would be problematic any existing nodes (their pod would be recreated with a different hostPath mount).


My only other idea would be to replace the DaemonSet with a static pod manifest that kubelet runs. Nodeup creates the kube-proxy static pod manifest and would be able to conditionally set the hostPath based on the distro since it is being set at node-launch time rather than as a DaemonSet. I'm sure thats not ideal though given how far we'd be deviating from calico's suggested deployment manifests. 

I'll try to think of other ideas...